### PR TITLE
Complete Task 7 POST /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by Dimeben**

### CODE REVIEW

File: models/articles-model.js [hunk 1]
```diff
@@ -1,54 +1,34 @@
 const db = require("../db/connection");
 
 exports.selectArticles = (articleId) => {
-  if (!Number(articleId)) {
+  if (!Number.isInteger(Number(articleId))) {
```
[the remark this answers is not in the document]
Comment (nc-verity-gregory) [reply models/articles-model.js:4]: This function still implies several articles when in fact you are selecting one.

File: models/comments-model.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const db = require("../db/connection");
+const { selectArticles } = require("../models/articles-model");
+
+exports.selectComments = (articleId) => {
+  if (!Number.isInteger(Number(articleId))) {
```
Comment (nc-verity-gregory) [inline models/comments-model.js:5]: PSQL should be handling this, not a custom error handler.

File: models/comments-model.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  return db
+    .query(
+      "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC",
+      [articleId]
+    )
+    .then((result) => {
+      const comments = result.rows;
```
Comment (nc-verity-gregory) [inline models/comments-model.js:15]: It is not an error if an article has no comments. A new article wouldn't have any.

File: models/comments-model.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      if (comments.length === 0) {
+        return Promise.reject({ status: 404, msg: "Page not found" });
+      }
+      return comments;
+    });
+};
+
+exports.createComments = (articleId, { username, body }) => {
+  if (
+    !Number.isInteger(Number(articleId)) ||
```
Comment (nc-verity-gregory) [inline models/comments-model.js:25]: You have made specific error handlers for codes in psql but are rejecting yourself. If a query is using the wrong types as defined by the seed psql will reject for you.

File: models/comments-model.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    typeof username !== "string" ||
+    typeof body !== "string"
+  ) {
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  }
+
+  return selectArticles(articleId)
```
Comment (nc-verity-gregory) [inline models/comments-model.js:32]: In order to make sure the article exists, I'd recommend reading up on the notes. You will need to do this more than once _hint how can you do a 404 for comments??_ Consider a reusable function that checks if an article exists.

File: __tests__/articles.test.js [hunk 6]
```diff
@@ -82,7 +82,8 @@ describe("/api/articles", () => {
       .then((res) => {
         const articles = res.body.articles;
         articles.forEach((article) => {
-          expect(article).toHaveProperty("comment_count", expect.any(Number));
+          expect(article).toHaveProperty("comment_count");
+          expect(typeof Number(article.comment_count)).toBe("number");
```
[the remark this answers is not in the document]
Comment (Dimeben) [reply __tests__/articles.test.js:86]: Screenshot 2024-08-27 at 13 54 13
Screenshot 2024-08-27 at 13 58 03

[the remark this answers is not in the document]
Comment (Dimeben) [reply __tests__/articles.test.js:86]: Screenshot 2024-08-27 at 13 59 16
Screenshot 2024-08-27 at 13 58 03

File: __tests__/comments.test.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+      .then((res) => {
+        const comments = res.body.comments;
+        expect(comments).toHaveLength(11);
+        expect(Array.isArray(comments)).toBe(true);
+        comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id", expect.any(Number));
+          expect(comment).toHaveProperty("votes", expect.any(Number));
+          expect(comment).toHaveProperty("created_at", expect.any(String));
+          expect(comment).toHaveProperty("author", expect.any(String));
+          expect(comment).toHaveProperty("article_id", expect.any(Number));
```
[the remark this answers is not in the document]
Comment (Dimeben) [reply __tests__/comments.test.js:35]: Screenshot 2024-08-27 at 14 00 33
Screenshot 2024-08-27 at 14 00 50

File: __tests__/comments.test.js [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad request");
+      });
+  });
+  test("400 - POST - sends an appropriate status and error message when an invalid comment object is posted", () => {
+    const newComment = {
+      name: "rogersop",
+      comment: "What an interesting read!",
+      votes: 10000000000000,
+    };
```
[the remark this answers is not in the document]
Comment (Dimeben) [reply __tests__/comments.test.js:122]: Screenshot 2024-08-27 at 14 02 19
Screenshot 2024-08-27 at 14 02 31

File: error-handlers.js [hunk 9]
```diff
@@ -6,22 +6,30 @@ exports.psqlErrorHandler = (err, req, res, next) => {
...
 };
 
-exports.serverErrorHandler = (err, req, res, next) => {
-  res.status(500).send({ msg: "500: Internal server error!" });
+exports.invalidIdErrorHandler = (err, req, res, next) => {
+  if (err.code === "23503") {
+    res.status(404).send({ msg: "Page not found" });
+  } else {
+    next(err);
+  }
```
[the remark this answers is not in the document]
Comment (Dimeben) [reply error-handlers.js:22]: Screenshot 2024-08-27 at 14 09 22

File: models/articles-model.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dimeben) [reply models/articles-model.js:4]: Screenshot 2024-08-27 at 14 10 15
Screenshot 2024-08-27 at 14 10 29

File: models/comments-model.js (hunk 2 above)
Comment (Dimeben) [reply models/comments-model.js:5]: Screenshot 2024-08-27 at 14 11 06

File: models/comments-model.js (hunk 4 above)
Comment (Dimeben) [reply models/comments-model.js:25]: Screenshot 2024-08-27 at 14 14 14

**Pull request merged**